### PR TITLE
Remove index checks

### PR DESCRIPTION
**Pull request by seallard**
## Description
This PR removes index checks from the sample sheets, resolves https://github.com/Clinical-Genomics/cg/issues/2216.

Since we have removed cgstats we can also remove index checks from our sample sheets as discussed in https://github.com/Clinical-Genomics/cg/pull/2220 (closed to avoid merge conflicts).

### Changed
- Remove index checks from sample sheets

### This [version](https://semver.org/) is a
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

**Review by diitaz93 — APPROVED** [review]
Amazing 🌟 For the tests, I suggest generating the sample sheet and starting demultiplexing a flow cell. If the demultiplex does not raise an error in the first seconds of execution it is probably fine.

**Review by henrikstranneheim — APPROVED** [review]
Great 👍

File: cg/apps/demultiplex/sample_sheet/sample_sheet_creator.py [hunk 1]
```diff
@@ -125,11 +117,6 @@ def create_sample_sheet_content(self) -> List[List[str]]:
 
     def process_samples_for_sample_sheet(self) -> None:
         """Add dummy samples, remove unwanted samples and adapt remaining samples."""
```
Comment (Vince-janv) [inline cg/apps/demultiplex/sample_sheet/sample_sheet_creator.py:119]: Remove dummy samples from docstring

**Review by Vince-janv — APPROVED** [review]
This looks great! 🔥

**Comment by seallard** [comment]
Deployed to stage. Generated sample sheet successfully with `cg demultiplex samplesheet create` and submitted a demultiplexing job with `cg demultiplex flow-cell`.

**Pull request merged**